test/common/router/config_impl_speed_test.cc

```diff
@@ -132,13 +142,260 @@ static void bmRouteTableSizeWithExactPathMatch(benchmark::State& state) {
  *
  * This represents common OpenAPI path templating.
  */
-static void bmRouteTableSizeWithRegexMatch(benchmark::State& state) {
-  bmRouteTableSize(state, RouteMatch::PathSpecifierCase::kSafeRegex);
+static void bmRouteTableSizeWithRegexMatchFixPathLen(benchmark::State& state) {
+  bmRouteTableSize(state, RouteMatch::PathSpecifierCase::kSafeRegex,
+                  "^/shelves/[^\\\\/]+/route_", "/shelves/shelf_");
+}
+
+BENCHMARK(bmRouteTableSizeWithPathPrefixMatchFixPathLen)->RangeMultiplier(2)->Ranges({{1, 2 << 13}});
+BENCHMARK(bmRouteTableSizeWithExactPathMatchFixPathLen)->RangeMultiplier(2)->Ranges({{1, 2 << 13}});
+BENCHMARK(bmRouteTableSizeWithRegexMatchFixPathLen)->RangeMultiplier(2)->Ranges({{1, 2 << 13}});
+
+// Generate path rule prefix based on the pattern 'abcdefghij', and path_len
+static std::string GetPrefixPathRuleFromBase(int path_len) {
+  std::string path_rule_base = "abcdefghij"; // pattern
```

Comment:
Move into private namespace constexpr? go/totw/140#approaches-that-work-in-source-files